### ISSUE TITLE
Do not allow upgrading to the same code

### ISSUE THREAD

**Issue by pepyakin** [paritytech/cumulus#863]
https://github.com/paritytech/polkadot/pull/4457 will impose an additional constraint on upgrading: the parachain won't be able to upgrade to the same code. While this is silly, if attempted, this will most likely break the parachain. Thus parachain-system should protect the users from doing that.

**Comment by bkchr** [comment]
Why will that break the parachain? Why isn't that just rejected by the relay chain if it doesn't want this? Just send the signal that the upgrade was rejected?

**Comment by pepyakin** [comment]
Yeah, this is exactly the problem: the relay-chain will reject such an upgrade. Parachain will break because it won't receive a `GoAhead::Abort` signal. The relay-chain does not the abort signal because of technical reasons: right now the code structured in such a way, that signals set in `schedule_code_upgrade` will be removed in `note_new_head`. I pitched an idea to reverse their order but was told that it is too big of a change.

We can try to work on this issue more, but I figured it would be just easier to forbid this.

**Comment by pepyakin** [comment]
Ah, that's right, stall.

Which makes me think that we should add some recovery functions for that as well.
